test/tests/tag.js

```diff
@@ -75,4 +77,67 @@ describe("Tag", function() {
         assert.equal(tagNames.length, 1);
       });
   });
+
+  it("can create a new annotated tag in a repo and delete it", function() {
+    var oid = Oid.fromString(commitPointedTo);
+    var name = "created-annotated-tag";
+    var repository = this.repository;
+
+    return repository.createTag(oid, name, tagMessage)
+      .then(function(tag) {
+        testTag(tag, name);
+      })
+      .then(function() {
+        return repository.createTag(oid, name, tagMessage)
+      })
+      .then(function() {
+        return Promise.reject(new Error("should not be able to create the '" + name + "' tag twice"));
+      }, function() {
+        return Promise.resolve();
+      })
+      .then(function() {
+        return repository.deleteTagByName(name);
+      })
+      .then(function() {
+        return Reference.lookup(repository, "refs/tags/" + name);
+      })
+      .then(function() {
+        return Promise.reject(new Error("the tag '" + name + "' should not exist"));
+      }, function() {
+        return Promise.resolve();
+      });
+  });
```

Comment:
I don't really like that this test bundles both the creation and deletion together into a single test but if the tag isn't cleaned up the tests aren't repeatable.